src/frontend/components/ScrollToTopButton.js

```diff
@@ -0,0 +1,54 @@
+import CatImage from '../assets/cat.png';
+import UpIcon from '../assets/up.svg'
+
+export default class ScrollToTopbutton {
+    constructor({ $parent }) {
+        this.$target = document.createElement('div');
+        this.$target.classList.add("scroll-top-button")
+
+        $parent.appendChild(this.$target);
+
+        this.render();
+        this.handleEvent();
+    }
+
+    handleEvent() {
+        const throttle = (callback, delay) => {
+            let timerId = null;
+            return (event) => {
+                if (timerId) return;
+                timerId = setTimeout(() => {
+                    callback(event);
+                    timerId = null;
+                }, delay);
+            };
+        };
+
+        const handleScroll = () => {
+            const rootElement = document.documentElement
+            const scrollTotal = rootElement.scrollHeight - rootElement.clientHeight
+
+            if ((rootElement.scrollTop / scrollTotal) > 0.10) {
+                this.$target.classList.add('show-button')
+            } else {
+                this.$target.classList.remove('show-button')
+            }
```

Comment:
```suggestion
            if ((rootElement.scrollTop / scrollTotal) > 0.10) {
                this.$target.classList.add('show-button')
                return
            }
            this.$target.classList.remove('show-button')
```
early return을 사용해서 indent depth를 줄이면 가독성이 더 좋아질 것 같습니다 :)